task/ecosystem-cert-preflight-checks/0.2/ecosystem-cert-preflight-checks.yaml

```diff
@@ -213,6 +219,40 @@ spec:
           value: "$(steps.generate-container-auth.results.auth-json-path)"
         - name: PFLT_KONFLUX
           value: "true"
+        - name: PARAM_IMAGE_URL
+          value: $(params.image-url)
+        - name: PARAM_PLATFORM
+          value: $(params.platform)
+      script: |
+        #!/usr/bin/env bash
+
+        set -o errexit
+        set -o nounset
+        set -o pipefail
+
+        image_url="${PARAM_IMAGE_URL}"
+        platform="${PARAM_PLATFORM}"
+
+        if [ -n "$platform" ]; then
+          # Extract part after slash if present
+          arch="${platform#*/}"
+          if [ "$arch" = "x86_64" ]; then
+            arch="amd64"
+          fi
+
+          # Validate against supported arch list. If it's not a known arch, fallback to amd64
+          case "$arch" in
+            amd64|ppc64le|arm64|s390x)
+              ;;
+            *)
+              arch="amd64"
+              ;;
+          esac
+
+          /usr/local/bin/preflight check container "$image_url" --platform "$arch"
```

Comment:
Do we need to handle the case where `$arch` isn't set in case there is a malformed platform? Maybe we should return an error result if we cannot parse the platform properly?

Reply:
Currently, we silently fall back to amd64 for unknown or malformed platforms.
Do you suggest that we should return an error result instead of falling back in those cases?

Reply:
If you want to fall back, then I wouldn't do it silently. This would prevent issues where users are missing a digest from being processed unknowingly

Reply:
Updated. Instead of  falling back to `amd64` silently, an error result is returned for malformed or unsupported architecture.